src/raft.c

```diff
@@ -1620,23 +1620,19 @@ void replaceShardGroups(RedisRaftCtx *rr, raft_entry_t *entry)
     ShardingInfo *si = rr->sharding_info;
 
     if (si->shard_group_map != NULL) {
+        void *data;
```

Comment:
Better to keep it in the right type - `ShardGroup *`

Reply:
sure